File: tree-document-model-js/TreeRoot.ts
import {HierarchyConfiguration} from "./HierarchyConfiguration";
import {TreeNode} from "./TreeNode";
import {IconConfiguration} from "./IconConfiguration";
import {ActionConfiguration} from "./ActionConfiguration";
import {OrderedSet} from "crossecore/lib/OrderedSet";
import {Bag} from "crossecore/lib/Bag";
import {InternalEObject} from "crossecore/lib/InternalEObject";

export interface TreeRoot
extends InternalEObject

{
	id:string;
	
	rootNode:TreeNode;
	actions:ActionConfiguration;
	hierarchy:HierarchyConfiguration;
	icons:IconConfiguration;
	

}

File: model-server/src/main/java/no/ntnu/stud/krirek/treelsp/model/tree/TreeRoot.java
package no.ntnu.stud.krirek.treelsp.model.tree;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
public interface TreeRoot {

    @NotNull String id();
    @Nullable TreeNode rootNode();
    @NotNull ActionConfiguration actions();
    @NotNull HierarchyConfiguration hierarchy();
    @Nullable IconConfiguration icons();
}
